sdk/monitor/azure-monitor-query/azure/monitor/query/_helpers.py

```diff
@@ -49,26 +50,34 @@ def order_results(request_order, responses):
     ordered = [mapping[id] for id in request_order]
     return ordered
 
-def construct_iso8601(start=None, end=None, duration=None):
+def construct_iso8601(timespan=None):
+    if not timespan:
+        return None
+    try:
+        start, end, duration = None, None, None
+        if isinstance(timespan[1], datetime):
+            start, end = timespan[0], timespan[1]
+        elif isinstance(timespan[1], timedelta):
+            start, duration = timespan[0], timespan[1]
+        else:
+            raise ValueError('Tuple must be a start datetime with a timedelta or an end datetime.')
+    except TypeError:
+        duration = timespan
     if duration is not None:
-        duration = 'PT{}S'.format(duration.total_seconds())
+        try:
+            duration = 'PT{}S'.format(duration.total_seconds())
+        except AttributeError:
+            raise ValueError('timespan must be a timedelta or a tuple.')
     iso_str = None
     if start is not None:
         start = Serializer.serialize_iso(start)
-        if end and duration:
-            raise ValueError("start_time can only be provided with duration or end_time, but not both.")
         if end is not None:
             end = Serializer.serialize_iso(end)
             iso_str = start + '/' + end
         elif duration is not None:
             iso_str = start + '/' + duration
         else:
```

Comment:
In which case will enter here?

Reply:
premeptively raising  in case start is provided with something other than a timedelta or a datetime

Reply:
It looks to me here is the case start is not None while end and duration are None?

Reply:
improved the message to warn about none explicitly